### ISSUE TITLE
[FEATURE] Create @NotBlank annotation for null and blank checks on string values

### ISSUE THREAD

**Issue by MattMCloudy** [rzwitserloot/lombok#2627]
> **Before submitting**
Please check our [wiki](https://github.com/rzwitserloot/lombok/wiki) because some feature requests are asked _a lot_ and there are good reasons (as explained on the wiki) why we can't (yet) implement them or why we won't implement them.

I looked through the wikis and did not see any posts referencing my use case. I did read the section about closing issues without reason, I certainly understand if that is the conclusion here.

**Feature Description**
When composing internal business models for various features, we find ourselves using lombok extensively i.e.

```
@Builder
@AllArgsConstructor
public class InternalModel {

    @NonNull
    public final String nonNullString;
    
}
```

However, in almost all cases where we would like to validate that the String is non-null, we would also like to validate that the string is not blank. A solution to this could be an `@NotBlank` annotation that throws an exception (either NPE or IllegalArg by default) when a String parameter, field, or local variable is null or blank. Currently, in places where we would use the `@NonNull` annotation we find ourselves defining our own constructor with non-null and not-blank validations.

It would certainly be preferable to maintain entirely lombok-based model's classes for the ease of readability and maintainability.

**Target Audience**
This appears, to me, to be rather general use. String values are used in almost all Java applications.

**Additional context**
Depending on the level of interest for this, I may be willing to write a PR. Let me know if you feel this is necessary or have another, better solution that you would recommend for our use case.

**Comment by janrieke** [comment]
Duplicate oft #646

**Comment by MattMCloudy** [comment]
Agreed it does seem similar. If you’d like to track these variants of emptiness checks in the same issue feel free to close.

**Comment by krabat** [comment]
I have the same requirement in many projects, so this feature would be very useful.

As outlined in my [comment](https://github.com/projectlombok/lombok/issues/646#issuecomment-1413374832) in issue  #646 it would be great if trimming of the String would optionally be supported, e.g.:

```
//content is not null and not empty after trimming
@NonBlank
String s;
```

```
//content is not null and not empty
@NonBlank( trim=false )
String s;
```
